docs/BOOT_LOADER_SPECIFICATION.md

```diff
@@ -87,6 +87,8 @@ Each configuration drop-in snippet must include at least a `linux` or an `efi` k
 
 On EFI systems all kernel images shall be EFI images. In order to be compatible with EFI systems it is highly recommended only to install EFI kernel images, even on non-EFI systems, if that's applicable and supported on the specific architecture.
 
+For maximum compatibility, it is highly recommended only to use kernel images that don't rely on BIOS services such as legacy BIOS calls or UEFI Boot Services.
```

Comment:
This reads as contradictory to the previous paragraph. Maybe clear that up explicitly.

Maybe: "Conversly, in order to increase maximum compatibility it is recommended to install generic kernel images that make little hard assumptions about the firmware they run on, i.e. it is a good idea that both images shipped as UEFI PE images and those which are not, don't make unnecessary assumption on the underlying firmware, i.e. don't hard depend on legacy BIOS calls or UEFI boot services." – or something like that.